plugin/src/main/java/org/opensearch/ml/action/connector/GetConnectorTransportAction.java

```diff
@@ -84,62 +80,48 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLConn
             .build();
         User user = RestActionUtils.getUserContext(client);
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            sdkClient
-                .getDataObjectAsync(getDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
-                .whenComplete((r, throwable) -> {
-                    context.restore();
-                    log.debug("Completed Get Connector Request, id:{}", connectorId);
-                    if (throwable != null) {
-                        Throwable cause = throwable.getCause() == null ? throwable : throwable.getCause();
-                        if (cause instanceof IndexNotFoundException) {
-                            log.error("Failed to get connector index", cause);
-                            actionListener.onFailure(new OpenSearchStatusException("Failed to find connector", RestStatus.NOT_FOUND));
-                        } else {
-                            log.error("Failed to get ML connector {}", connectorId, cause);
-                            actionListener.onFailure(new RuntimeException(cause));
-                        }
-                    } else {
-                        if (r != null && r.parser().isPresent()) {
-                            try {
-                                XContentParser parser = r.parser().get();
-                                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-                                Connector mlConnector = Connector.createConnector(parser);
-                                if (!TenantAwareHelper
-                                    .validateTenantResource(mlFeatureEnabledSetting, tenantId, mlConnector.getTenantId(), actionListener)) {
-                                    return;
-                                }
-                                mlConnector.removeCredential();
+            connectorAccessControlHelper
+                .getConnector(
```

Comment:
Not specific to this PR but we should have a task to cache connector and models. Fetching connector/model for each predict request will increase the latency of predict request.